python_src/evolution/random_evolution.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from deap import tools, creator, base
+from threading import Thread, Event
+
+# Event object used to send signals from one thread to another
+STOP_EVENT = Event()
+FIRST = 0
+
+
+class RandomEvolution:
+    def __init__(self, orig_ind, fitness, generate_random_ind, select, mutate, expectations):
+        creator.create("FitnessMax", base.Fitness, weights=(1.0,))
+        creator.create("Individual", list, fitness=creator.FitnessMax)
+
+        self.toolbox = base.Toolbox()
+        # Attribute generator
+        self.toolbox.register("random_ind", generate_random_ind, orig_ind, params={"min": 10, "max": 50, "dim": 1})
+        # Structure initializers
+        self.toolbox.register("individual", tools.initRepeat, creator.Individual, self.toolbox.random_ind, 1)
+        self.toolbox.register("population", tools.initRepeat, list, self.toolbox.individual)
+
+        self.toolbox.register("evaluate", fitness)
+        self.toolbox.register("mutate", mutate, mutate_params={"std": 0.25, "dim": 1, "min": 10, "max": 50})
```

Comment:
Same as above. Is max 50 a suitable value? I also suggest we factor those parameters into variables, so mutation and random generation uses exactly the same limits.

Also, since this is the random generation, why do we need a mutate operation? The idea of random is to generate a new random individual at each iteration, not mutate existing individuals.

Reply:
Got it. The mutate function was removed.